backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/ExternalSystemConfigRepo.java

```diff
@@ -74,6 +76,24 @@ public ExternalSystemParentConfig getById(final @NonNull IExternalSystemChildCon
 		}
 	}
 
+	public Optional<IExternalSystemChildConfig> getChildByParentIdAndType(@NonNull final ExternalSystemParentConfigId id, @NonNull final ExternalSystemType externalSystemType)
```

Comment:
why do we need the type? it's stored in the `EXternalSystem_Config` after all.

Reply:
I was thinking that it is better in terms of performance since you don't have to fetch the parent to get the type or fetch for every type.